tuning/data/setup_dataprocessor.py

```diff
@@ -107,15 +107,21 @@ def _get_pretokenized_dataset_handlers(data_args, packing, is_eval_tokenized):
 
 
 ### Data format 2
-def _get_dataset_formatting_handlers(data_args, packing):
+def _get_dataset_formatting_handlers(data_args, packing, is_padding_free=False):
 
     if data_args.response_template is None:
         if packing is False:
-            raise ValueError(
-                "Since dataset_text_field or data_formatter_template \
-                is provided and packing is disabled, \
-                needs a corresponding response template for masking"
-            )
+            if is_padding_free:
+                logger.debug(
+                    "Assuming extended pretraining scenario because, packing is false,"
+                    + " padding_free plugin is used and no response template was provided."
+                )
+            else:
+                raise ValueError(
+                    "Since dataset_text_field or data_formatter_template \
+                    is provided and packing is disabled, \
+                    needs a corresponding response template for masking"
+                )
```

Comment:
We can support a pretraining scenario (predict all tokens) in this case as well however, it wont be efficient due to involvement of padding. We can use 

```python
DataCollatorForSeq2Seq(
                tokenizer=tokenizer, padding=True, max_length=max_seq_length
            )
```

However, if we are not planning to encourage this flow its good to capture that well in the ValueError encouraging them to use either padding_free or packing=True

Reply:
It is a valid case to use padding free with packing.